shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/AbstractShenyuPlugin.java

```diff
@@ -79,76 +79,72 @@ public abstract class AbstractShenyuPlugin implements ShenyuPlugin {
-        if (Objects.nonNull(pluginData) && pluginData.getEnabled()) {
-            final String path = exchange.getRequest().getURI().getPath();
-            SelectorData selectorData = obtainSelectorDataCacheIfEnabled(exchange);
+        // early exit
+        if (Objects.isNull(pluginData) || !pluginData.getEnabled()) {
+            return chain.execute(exchange);
+        }
+        final String path = exchange.getRequest().getURI().getPath();
+        List<SelectorData> selectors = BaseDataCache.getInstance().obtainSelectorData(pluginName);
+        SelectorData selectorData = obtainSelectorDataCacheIfEnabled(path);
+        // handle Selector
+        if (Objects.nonNull(selectorData) && StringUtils.isBlank(selectorData.getId())) {
+            return handleSelectorIfNull(pluginName, exchange, chain);
+        }
+        if (Objects.isNull(selectorData)) {
+            if (CollectionUtils.isEmpty(selectors)) {
+                return handleSelectorIfNull(pluginName, exchange, chain);
+            }
+            Pair<Boolean, SelectorData> matchSelectorData = matchSelector(exchange, selectors);
+            selectorData = matchSelectorData.getRight();
             if (Objects.isNull(selectorData)) {
-                List<SelectorData> selectors = BaseDataCache.getInstance().obtainSelectorData(pluginName);
-                if (CollectionUtils.isEmpty(selectors)) {
-                    return handleSelectorIfNull(pluginName, exchange, chain);
-                }
-                Pair<Boolean, SelectorData> matchSelectorData = matchSelector(exchange, selectors);
-                selectorData = matchSelectorData.getRight();
-                if (Objects.isNull(selectorData)) {
-                    if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
-                        selectorData = new SelectorData();
-                        selectorData.setPluginName(named());
-                        cacheSelectorData(path, selectorData);
-                    }
-                    return handleSelectorIfNull(pluginName, exchange, chain);
-                } else {
-                    if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
-                        cacheSelectorData(path, selectorData);
-                    }
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
+                    selectorData = new SelectorData();
+                    selectorData.setPluginName(pluginName);
+                    cacheSelectorData(path, selectorData);
                 }
+                return handleSelectorIfNull(pluginName, exchange, chain);
             } else {
-                if (StringUtils.isBlank(selectorData.getId())) {
-                    return handleSelectorIfNull(pluginName, exchange, chain);
+                if (matchCacheConfig.getSelectorEnabled() && matchSelectorData.getLeft()) {
```

Comment:
alreay cache，not again cache

Reply:
this is from From the old logic, seem No duplicate cache appears

![image](https://user-images.githubusercontent.com/12622645/202112154-fccf07a7-f7e6-49ed-ad98-f288328daad3.png)

![image](https://user-images.githubusercontent.com/12622645/202113037-fa62e8a5-2a93-4dad-b459-d4641157126a.png)